### DISCUSSION TITLE
Adding a heavy node to a fully distributed setup

### DISCUSSION THREAD

**Discussion by presianbg** [Security-Onion-Solutions/securityonion#6310] [Unsupported Versions] [answered]
Hi,

TL:DR - Is it OK to add a heavy node to distributed setup ? 

Our setup has one manager node, few search nodes and several forward nodes.
We need a heavy node to sore the data locally, but I'm wondering if doing so will  influence the whole distributed cluster in a negative way (performance degradation or configuration incompatibilities).

Cheers,
PY

**Comment by dougburks** [comment, marked as answer]
From https://docs.securityonion.net/en/2.3/architecture.html#heavy-node:
> Heavy nodes are NOT recommended for most users.

In theory, you could add a heavy node to a distributed deployment. However, there can be performance degradation as anytime you do a search you are searching across your search nodes and the heavy node and the manager must wait for all results from all nodes.
